### PR TITLE
Using the new Jenkins client library

### PR DESCRIPTION
**Pull request by LinuxSuRen**
### What type of PR is this?

/kind chore

### What this PR does / why we need it:
We intend to use [jenkins-zh/jenkins-client](https://github.com/jenkins-zh/jenkins-client) to replace current XML based Jenkins client.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:

You can use the following image to test this PR:
```
surenpi/devops-controller:dev-v3.2.1-f7661f6
```

### Does this PR introduce a user-facing change??

```release-note
None
```

This PR will replace https://github.com/kubesphere/ks-devops/pull/313

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
Before the upstream PR is https://github.com/jenkins-zh/jenkins-client/pull/43 got merged, we need to hold it.

/hold

File: controllers/pipeline/pipeline_controller_test.go [hunk 1]
```diff
@@ -345,15 +354,31 @@ func TestCreatePipeline(t *testing.T) {
...
 	pipelineName := "test"
 	projectName := "test_project"
-
+	spec := devops.PipelineSpec{
+		Type: devops.NoScmPipelineType,
+		Pipeline: &devops.NoScmPipeline{
+			Name: pipelineName,
+		},
+	}
+	//createPayload := job.CreateJobPayload{
```
Comment (JohnNiang) [inline controllers/pipeline/pipeline_controller_test.go:363]: Please remove unused lines.

**Review by JohnNiang — CHANGES_REQUESTED** [review]
All changes look good to me currently, but the lines commented.

File: controllers/pipeline/pipeline_controller_test.go (hunk 1 above)
Comment (LinuxSuRen) [reply controllers/pipeline/pipeline_controller_test.go]: Sure.

**Review by JohnNiang — APPROVED** [review]

**Pull request merged**